.github/publish-test.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+# This is a script meant to be used by Github Actions to test out pubsys related cargo-make tasks
+
+required_env() {
+  local env="${1:?}"
+  local value="${2}"
+  if [ -z "${value}" ]; then
+    echo "ERROR: Environment variable ${env} is required to be set" >&2
+    exit 2
+  fi
+}
+
+required_env "VARIANT" "${VARIANT}"
+required_env "ARCH" "${ARCH}"
+required_env "GITHUB_WORKSPACE" "${GITHUB_WORKSPACE}"
+
+echo -e "\n=^..^=   =^..^=   =^..^=   Create TUF repository from Bottlerocket build artifacts   =^..^=   =^..^=   =^..^=\n"
+cargo make -e BUILDSYS_VARIANT="${VARIANT}" -e BUILDSYS_ARCH="${ARCH}" repo
+mv build/repos/default build/repos/test
+
+cat <<EOF > Infra.toml
+[repo.default]
+metadata_base_url = "file://${GITHUB_WORKSPACE}/build/repos/test/latest"
+targets_url = "file://${GITHUB_WORKSPACE}/build/repos/test/latest/targets"
+EOF
+
+echo -e "\n=^..^=   =^..^=   =^..^=   Test check-repo-expirations   =^..^=   =^..^=   =^..^=\n"
+cargo make -e BUILDSYS_VARIANT="${VARIANT}" -e BUILDSYS_ARCH="${ARCH}" check-repo-expirations
+echo -e "\n=^..^=   =^..^=   =^..^=   Test validate-repo   =^..^=   =^..^=   =^..^=\n"
+cargo make -e BUILDSYS_VARIANT="${VARIANT}" -e BUILDSYS_ARCH="${ARCH}" validate-repo
+echo -e "\n=^..^=   =^..^=   =^..^=   Test refresh-repo   =^..^=   =^..^=   =^..^=\n"
+cargo make -e BUILDSYS_VARIANT="${VARIANT}" -e BUILDSYS_ARCH="${ARCH}" refresh-repo
+
+REPO_PATH=$(ls -d -1 ${GITHUB_WORKSPACE}/build/repos/default/* | head -n 1)
```

Comment:
```suggestion
# refresh-repo doesn't change the `latest` symlink, so we find the new directory.
REPO_PATH=$(find build/repos/default/ -mindepth 1 -maxdepth 1 -type d)
```
`ls` can be dangerous for finding files, and I'm not sure how we can guarantee that we wouldn't get `latest/` from the `ls` - I think you said `latest/` wouldn't work because the link isn't updated by refresh-repo.  This would ensure we only get the one _directory_ in `default/`.

Reply:
The `latest` symlink won't get created. So the directory after a refresh-repo would look like this:

```
$ tree build/repos/default/
build/repos/default/
└── bottlerocket-1.1.3-be591681
    └── aws-k8s-1.19
        └── aarch64
            ├── 1626372511.snapshot.json
            ├── 1626372511.targets.json
            ├── 1.root.json
            └── timestamp.json

3 directories, 4 files

```

Reply:
Ah, ok.  I do think the find is a bit safer, since it wouldn't raise any `ls` concerns if we change the structure in the future.